middleware.ts

```diff
@@ -2,28 +2,67 @@ import { getIronSession } from 'iron-session';
 import type { NextRequest } from 'next/server';
 import { NextResponse } from 'next/server';
 import { ISession } from '#models/user/session';
-import { extractSirenOrSiretSlugFromUrl } from '#utils/helpers';
+import {
+  extractSirenOrSiretSlugFromUrl,
+  isLikelyASiretOrSiren,
+} from '#utils/helpers';
 import { sessionOptions, setVisitTimestamp } from '#utils/session';
 
-// This function can be marked `async` if using `await` inside
-export async function middleware(request: NextRequest) {
-  const pathname = request.nextUrl.pathname;
-  const sirenOrSiretSlug = extractSirenOrSiretSlugFromUrl(pathname);
+const shouldRedirect = (path: string, search: string, url: string) => {
```

Comment:
I think it's a good thing to have all the redirect logic in the same place instead of scattered through the app, with exceptions thrown from the model.